packages/core/src/credentials/issuer.ts

```diff
@@ -0,0 +1,150 @@
+ */
+
+import type { Did, ICType, IClaimContents } from '@kiltprotocol/types'
+
+import { SDKErrors } from '@kiltprotocol/utils'
+import { KiltAttestationProofV1, KiltCredentialV1 } from './V1/index.js'
+import type { UnsignedVc, VerifiableCredential } from './V1/types.js'
+import type { CTypeLoader } from './ctype/index.js'
+import type { IssuerOptions } from './interfaces.js'
+
+/**
+ * Creates a new credential document as a basis for issuing a credential.
+ * This document can be shown to users as a preview or be extended with additional properties before moving on to the second step of credential issuance:
+ * Adding a `proof` to the document using the {@link issue} function to make the credential verifiable.
+ *
+ * @param arguments Object holding all arguments for credential creation.
+ * @param arguments.type A type string identifying the (sub-)type of Verifiable Credential to be created.
+ * This is added to the `type` field on the credential and determines the `credentialSchema` as well.
+ * Defaults to the type {@link KiltCredentialV1.CREDENTIAL_TYPE KiltCredentialV1} which, for the time being, is also the only type supported.
+ * @param arguments.issuer The Decentralized Identifier (DID) of the identity acting as the authority issuing this credential.
+ * @param arguments.subject The DID identifying the subject of the credential, about which `claims` are made.
+ * Sets the `credentialSubject.id` property of the credential.
+ * @param arguments.claims An object containing key-value pairs that represent claims made about the subject.
+ * These key-value pairs are added to the `credentialSubject` property of the credential.
+ * @param arguments.cType CTypes are special credential subtypes that are defined by a schema describing claims that may be made about the subject and are registered on the Kilt blockchain.
+ * Each Kilt credential is based on exactly one of these subtypes. This argument is therefore mandatory and expects the schema definition of a CType.
+ * @param arguments.cTypeDefinitions Some CTypes are themselves composed of definitions taken from other CTypes; in that case, these definitions need to be supplied here.
+ * Alternatively, you can set a {@link CTypeLoader} function that takes care of fetching all required definitions.
+ * @param arguments.legitimations Any credentials that the issuer uses to provide legitimations for their authority for certifying the claims made.
+ * @returns A (potentially only partial) credential that is yet to be finalized and made verifiable with a proof.
+ */
+export async function createCredential({
+  type,
+  issuer,
+  subject,
+  claims = {},
+  cType,
+  cTypeDefinitions,
+  legitimations = [],
+}: {
+  type?: string
+  issuer: Did
+  subject: Did
+  claims?: Record<string, unknown>
+  cType: ICType
+  cTypeDefinitions?: ICType[] | CTypeLoader
+  // TODO: shall we support legitimations and delegations here or are these advanced features you'd have to dig for?
```

Comment:
I think we can consider these after our first user actually uses them.

Reply:
Yeah I don't think it would be an issue to add them later if we wanted to. My main argument against was that given no-one uses them, we decide not to include them in future credential/proof versions, or replace them with related solutions that are being floated in the VC space.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/811/commits/1641d6c27df043f959355f5dbbbf46390608645a

Reply:
Did you make a ticket for this?

Reply:
mhhh it'll be one of those dead tickets that we keep pushing. We'll remember if it becomes relevant.